### PR TITLE
Make compatible with Wagtail >= 2.2 (issue #13)

### PR DESCRIPTION
**Pull request by stefan-squareweave**

### CODE REVIEW

File: wagtailmodelchooser/static/wagtailmodelchooser/js/model_chooser.js [hunk 1]
```diff
@@ -1,11 +1,101 @@
...
+        var dataObj = {ajax: 1}
+        if (window.instance) dataObj['instance'] = window.instance;
+        if (searchTerm) dataObj['q'] = searchTerm;
+        if (page) dataObj['p'] = page;
+
+        $.ajax({
+            url: searchUrl,
+            data: dataObj,
+            success: function(data, status) {
+                console.log(data, status);
```
Comment (pierremanceaux) [inline wagtailmodelchooser/static/wagtailmodelchooser/js/model_chooser.js]: Does this console.log have to stay?

File: wagtailmodelchooser/static/wagtailmodelchooser/js/model_chooser.js [hunk 2]
```diff
@@ -14,11 +104,15 @@ function createModelChooser(id, url) {
...
-        modal.body.on("submit", "form", function (e) { e.preventDefault(); });
     });
 
     $('.action-clear', chooserElement).click(function() {
         input.val('');
         chooserElement.addClass('blank');
     });
 }
+
+// Register a global function to be called when our model chooser widget
```
Comment (pierremanceaux) [inline wagtailmodelchooser/static/wagtailmodelchooser/js/model_chooser.js]: Maybe keep consistency with your comments style and use `/* */`?

**Comment by seb-b** [comment]
Hi @stefan-squareweave 	

Thanks for this! Haven't had to use the package with 2.2 yet. tox + setuptools stuff all looks good.

I'm not familiar with how the new modal code works on wagtail side. Why was the name change from `modal.js` to `modal_chooser.js` needed and how is that static file included in the template? I can see any reference to the new filename anywhere

**Comment by stefan-squareweave** [comment]
Force pushed fixes for the above code-review comments.

@seb-b The modal_chooser.js file was already included to admin via [wagtail_hooks.py](https://github.com/takeflight/wagtailmodelchooser/blob/master/wagtailmodelchooser/wagtail_hooks.py) .

Wagtail 2.2 reworked the render_modal_workflow function so you can no longer inject javascript to the page via that call. In the new regime the ModalWorkflow system needs to already be registered with the Wagtail block before you click the button. We do this in render_js_init in the widgets.py file. The button click then just triggers this already registered workflow by name using the 'step' json_data argument (and some XHR magic I didn't look in to).

**Pull request merged**